### PR TITLE
Add MongoDB GridFS native support

### LINKED ISSUE #932: mongo-gridfs should reuse the Quarkus Mongo extension

### PR DESCRIPTION
**Pull request by jamesnetherton**
Fixes #932

### CODE REVIEW

File: extensions-support/mongodb/deployment/src/main/java/org/apache/camel/quarkus/support/mongodb/deployment/SupportMongoDBProcessor.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        for (MongoClientBuildItem mongoClient : mongoClients) {
+            // If there is a default mongo client instance, then bind it to the camel registry
+            // with the default mongo client name used by the camel-mongodb component
+            if (MongoClientRecorder.DEFAULT_MONGOCLIENT_NAME.equals(mongoClient.getName())) {
+                runtimeBeans.produce(
+                        new CamelRuntimeBeanBuildItem(
+                                "camelMongoClient",
+                                "com.mongodb.client.MongoClient",
+                                mongoClients.get(0).getClient()));
```
Comment (ppalaga) [inline extensions-support/mongodb/deployment/src/main/java/org/apache/camel/quarkus/support/mongodb/deployment/SupportMongoDBProcessor.java:50]: Maybe better this?

```suggestion
                                mongoClient.getClient()));
```

File: extensions-support/mongodb/runtime/pom.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.camel.quarkus</groupId>
+        <artifactId>camel-quarkus-support-mongodb-parent</artifactId>
+        <version>1.1.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>camel-quarkus-support-mongodb</artifactId>
+    <name>Camel Quarkus :: Support :: MongoDB :: Runtime</name>
```
Comment (ppalaga) [inline extensions-support/mongodb/runtime/pom.xml:30]:
```suggestion
    <name>Camel Quarkus :: Support :: MongoDB :: Runtime</name>
    <description>Camel Quarkus Support MongoDB</description>
```

File: extensions-support/mongodb/runtime/pom.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    <properties>
+        <firstVersion>1.0.0-M8</firstVersion>
```
Comment (ppalaga) [inline extensions-support/mongodb/runtime/pom.xml:33]:
```suggestion
        <firstVersion>1.0.0-M8</firstVersion>
        <quarkus.metadata.unlisted>true</quarkus.metadata.unlisted>
```

File: extensions-support/mongodb/deployment/src/main/java/org/apache/camel/quarkus/support/mongodb/deployment/SupportMongoDBProcessor.java (hunk 1 above)
Comment (jamesnetherton) [reply extensions-support/mongodb/deployment/src/main/java/org/apache/camel/quarkus/support/mongodb/deployment/SupportMongoDBProcessor.java]: Yes, good catch.

**Pull request merged**